### PR TITLE
Add a RestOAuth handler

### PR DESCRIPTION
**Pull request by klopyrev**
Add a RestOAuth handler that uses the REST server to generate and validate tokens, and that uses the users database for the get_users call. When we launch the new OAuth set up we'll need to update config.json to 

```
    "auth": {
        "class": "RestOAuthHandler"
    },
```

Note that with this setup token generation works the same as before. Token validation, however, is different. Since the REST server by default checks for a valid access token for every REST request to validate the token I've simply added an API that doesn't do much, aside from requiring request.user to be set. Finally, get_users doesn't get users from the OAuth server, instead using the local database, which is shared by the bundle service and the REST bundle service.

This change requires https://github.com/codalab/codalab-cli/pull/369 to be merged first.

Additionally, add a LocalUserAuthHandler that will be used to construct a LocalBundleClient to handle the legacy API calls migrated from Django to the Bottle app.

@percyliang @kashizui

### CODE REVIEW

**Comment by klopyrev** [comment]
Ugh, this change is broken. Don't review it yet.

**Comment by klopyrev** [comment]
Nevermind, please review this change.

File: codalab/model/bundle_model.py [hunk 1]
```diff
@@ -1468,23 +1468,40 @@ def get_user(self, user_id=None, username=None):
...
+            rows = connection.execute(select([
                 cl_user
-            ]).where(and_(*clauses)).limit(1)).fetchone()
-
-        if row is None or not row.is_active:
-            return None
+            ]).where(and_(*clauses))).fetchall()
 
-        return User(row)
+        return [User(row) for row in rows]
```
Comment (percyliang) [inline codalab/model/bundle_model.py:1504]: Weird, we weren't returning an array before...?

**Comment by percyliang** [comment]
LGTM

File: codalab/model/bundle_model.py (hunk 1 above)
Comment (klopyrev) [inline codalab/model/bundle_model.py:1504]: I updated the method to get multiple users at a time.

**Pull request merged**